Add integration tests for error messages from balance endpoint

piotr-iohk opened the pull request:
### Issue Number

ADP-1241
ADP-1225

### Comments

This PR adds:
- integration test coverage for the numeric/variable parts of a collateral selection error.
- integration test coverage for error messages from the balance point. (These are currently marked as pending, as we expect to improve these errors.)

Commit:
Integration tests for error messages from balance ep

Commit:
More tests against API returned errors

piotr-iohk commented on lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs:
```diff
@@ -1194,6 +1195,135 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
+            , expectErrorMessage errMsg403Fee
+            ]
+
+    it "TRANS_NEW_BALANCE_02c - Cannot balance on when I cannot afford collateral" $ \ctx -> runResourceT $ do
```
Possibly ADP-1241 case @jonathanknowles.

jonathanknowles replied:
@piotr-iohk I've prepared some enhancements to this test in the following branch: 
https://github.com/input-output-hk/cardano-wallet/compare/jonathanknowles/adp-1225/test-enhancements

Will make this into a PR tomorrow morning!

jonathanknowles replied:
Update, I've added these commits to this PR.

jonathanknowles committed:
Correct formatting of code and organize imports.

This follows the guideline here:

https://github.com/input-output-hk/adrestia/blob/master/process/eng-practices/Coding-Standards.md#limit-line-length-to-80-characters

jonathanknowles committed:
Correct wording within integration test titles.

jonathanknowles committed:
Use a more interesting UTxO distribution for `TRANS_NEW_BALANCE_02c`.

This will allow us to test the output of a failed collateral selection
where the largest available combination of pure ada UTxOs has more than
one entry.

jonathanknowles committed:
Test variable parts of `InsufficientCollateral` API error.

It's important to test the variable parts of this error so that we we
can verify that the user is being presented with sensible advice.

It's also important to verify the numerical values as a sanity check
that they are sensible.

Note that our property tests all use simulated cost models, so it's
doubly important to have /something/ in the integration test suite to
act as a check for sensible values.

jonathanknowles committed:
Improve rendering of `InsufficientCollateral` API error.

We:

- sort the largest available combination of pure ada entries into
  ascending order, which eases testing.
- use a more natural list format, which adds a space after the comma (to
  make the output more readable).

jonathanknowles committed:
Use smaller amounts of ada in `TRANS_NEW_BALANCE_02c`.

This is to account for recent fixes to the fee calculation logic within
the `balanceTransaction` function, in particular:

https://github.com/input-output-hk/cardano-wallet/pull/2989

jonathanknowles committed:
Mark failing integration tests (for balancing errors) as pending.

We expect to improve these errors in a future PR.
